src/main/java/Parser.java

```diff
@@ -1,8 +1,88 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
 /**
  * Deals with making sense of the user command.
  */
 public class Parser {
 
+    private static Optional<Command> findOneParameterCommand(String action) throws DukeException {
```

Comment:
Good use of Optional to help make the code cleaner. Able to also handle null values more efficiently